include/OpenColorIO/OpenColorIO.h

```diff
@@ -1521,15 +1522,21 @@ class OCIOEXPORT Config
      * trying to resolve all the FileTransforms in the Config to specific files is because of the
      * goal to allow context variables to continue to work.
      * 
+     * Archiving a minimal Config will try resolve these file references using the current context.
+     *
      * If a Config is created with CreateFromStream, CreateFromFile with an OCIOZ archive, or 
      * CreateFromConfigIOProxy, it cannot be archived unless the working directory is manually set 
      * to a directory that contains any necessary LUT files. 
      * 
      * The provided output stream must be closed by the caller, if necessary (e.g., an ofstream).
      *
      * \param ostream The output stream to write to.
+     * \param flags Flags top control archive creation
      */
-    void archive(std::ostream & ostream) const;
+    void archive(std::ostream & ostream, ArchiveFlags flags) const;
+
+    //TODO: document
+    void GetAllFileReferences(std::set<std::string> & files) const;
```

Comment:
That was one of my thoughts, I would think other uses for the list of files would be interesting and thus a public function may be needed.

Note that the existing code uses them as a set rather than a vector as this then makes each entry unique as it is obviously possible to use the same external file in multiple transforms in a config file (we use the same placeholder file for some of our configuration files). If we want to turn that into a vector for a public API then I guess that is fine.